pkg/controller/deployment_inplace.go

```diff
@@ -284,6 +289,13 @@ func (dc *controller) reconcileOldMachineSetsInPlace(ctx context.Context, allMac
 		return false, err
 	}
 
+	// Machines from old machine sets which are undergoing update will eventually move to new machine set.
+	// So once the current new machine set replcas + old machine set replicas undergoing update reaches the desired replicas,
+	// we can stop selecting machines from old machine sets for update.
+	if oldISsMachinesUndergoingUpdate+newMachineSet.Spec.Replicas >= deployment.Spec.Replicas {
+		return false, nil
```

Comment:
Should this not return `true, nil`? I assume we would want `syncRolloutStatus()` to run

Reply:
The last reconciliation would have already called the `syncRolloutStatus`, so we don't need it.